### ISSUE TITLE
SubStreams parameter causes SST and other engines to work badly

### ISSUE THREAD

**Issue by pnorbert** [ornladios/ADIOS2#2053]
**Describe the bug**
The BP engines have an engine parameter SubStreams. If left in the settings accidentally when using SST or other engines using BP serialization, they will not work properly on the reading side. It will run but the data will be corrupt on the reading side. 

**To Reproduce**
We see this behavior with nwchem sorter or gray-scott. 

**Additional context**
The problem lies in the core that allows parameters to be set that does not work for an engine but is (silently) used by subcomponents (BP serializer in this case).

**Comment by eisenhauer** [comment]
I was just looking at BPBase where this stuff is pulled out of the parameter list passed to Init.  So I think the question is how do we handle this?  Should we've got knowledge isolated in BP (what parameters it knows) but where there is compatibility issues with things elsewhere (e.g. SST).  SST sees the params first, and it could theoretically sanitize it, removing known incompatible parameter settings.  Or it could reach into BP*Serializer afterwards to set things to something safe.  But we could be asking for trouble when someone adds new BP parameters and doesn't know to possibly add it to the parameter sanitizers in other engines.  Maybe have something like a "RestoreSafeDefaults" method in BPBase that takes an engine name and unsets BP parameters as appropriate?  At least then we've got the BP parameter operations all in one place...  Other thoughts?
